### PR TITLE
Use iD presets as the source for this

### PR DESCRIPTION
**Pull request by bhousel**
This PR brings in the iD presets, fields, and build scripts.  If accepted, I'll remove these things from iD and make this repository the master source for iD's presets.

RE: openstreetmap/iD#2656

### CODE REVIEW

**Comment by pnorman** [comment]
is the iD format sufficiently rich to convert to the various editor formats? Should they be stored in here in some other format, and converted to the iD format and JOSM format by scripts?

File: README.md [hunk 1]
```diff
@@ -1,2 +1,98 @@
...
+directory hierarchy based on OSM key/value pairs. For example, the preset that matches
+the tag `leisure=park` is in the file `/presets/leisure/park.json`.
+
+## Preset Format
+
+A basic preset is of the form:
+
+```javascript
+{
+    // Maki icon which represents this feature.
```
Comment (pnorman) [inline README.md:18]: What about non-Maki icons?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,98 @@
...
+    "icon": "park",
+    // An array of field names.
+    "fields": [
+        "address"
+    ],
+    // The geometry types for which this preset is valid.
+    // options are point, area, line, and vertex.
```
Comment (pnorman) [inline README.md:25]: Can this notion of geometry be mapped to different editors?

**Comment by HolgerJeromin** [comment]
The diskussion on the josm dev ML is interesting about this. JOSM uses a more advanced preset system which can not generated from iD presets.

**Comment by bhousel** [comment]
> is the iD format sufficiently rich to convert to the various editor formats? Should they be stored in here in some other format, and converted to the iD format and JOSM format by scripts?

Somebody would have to define the format and write the scripts.  As I've said before, a new format solves a problem that nobody has.

**Comment by bhousel** [comment]
> The diskussion on the josm dev ML is interesting about this. JOSM uses a more advanced preset system which can not generated from iD presets.

Yes I've seen it, I don't think anybody is expecting JOSM to get involved in this effort - and their code and presets are license-incompatible anyway.  

Someone (I can't remember who) wrote scripts to turn the iD presets into JOSM ones - I expect that's the best way forward for users who want to use these presets in JOSM.

**Comment by bhousel** [comment]
I thought about it more and realized - I just did a dumb copy when I made this and instead probably should have done something a bit more clever to preserve the git history of the files. After discussion shakes out, I may redo this PR..

**Comment by pnorman** [comment]
If this is only going to work with iD, what's the point? The JOSM license isn't a blocker, as that stops this repo from using the existing JOSM presets, not this repo from producing JOSM preset files.

**Comment by HolgerJeromin** [comment]
This repository content can have the licence we want. Imo it would be possible to have josm presets with its license and use this even in iD. Vespucci uses different licenses for the code and the preset images. 
I don't see the that this should not be possible with id, too.

**Comment by pnorman** [comment]
> Imo it would be possible to have josm presets with its license and use this even in iD.

If we're using the MIT license for this repo, we can have presets for JOSM, but not the JOSM presets.

**Comment by bhousel** [comment]
Withdrawing this, as iD's presets have already moved significantly over the summer and there doesn't seem to be any interest.

**Comment by iandees** [comment]
Ack, I completely forgot about this PR Bryan. If you opened it up again
with more recent data I would probably merge it. I definitely have interest.

**Comment by bhousel** [comment]
> Ack, I completely forgot about this PR Bryan. If you opened it up again
> with more recent data I would probably merge it. I definitely have interest.

Ok cool, I'll resubmit when I have better availability..

**Pull request closed without merging**